From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Thu, 1 Nov 2018 16:47:24 -0700
Subject: [PATCH] Warn on use of insecure 3DES algo

---
 g10/encrypt.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/g10/encrypt.c b/g10/encrypt.c
index a96a779..e22c091 100644
--- a/g10/encrypt.c
+++ b/g10/encrypt.c
@@ -538,6 +538,15 @@ check_encryption_compliance (DEK *dek, pk_list_t pk_list)
   compliant = gnupg_cipher_is_compliant (CO_DE_VS, dek->algo,
                                          GCRY_CIPHER_MODE_CFB);
 
+  switch (dek->algo)
+  {
+    case GCRY_CIPHER_3DES:
+      log_error (_("WARNING: cipher algorithm '%s' is insecure and not"
+                   " suitable for encryption!\n"),
+                 openpgp_cipher_algo_name (dek->algo));
+      break;
+  };
+
   for (pkr = pk_list; pkr; pkr = pkr->next)
     {
       PKT_public_key *pk = pkr->pk;
@@ -886,6 +895,15 @@ encrypt_filter (void *opaque, int control,
 
           efx->cfx.dek->use_mdc = use_mdc (efx->pk_list,efx->cfx.dek->algo);
 
+          switch (efx->cfx.dek->algo)
+          {
+            case GCRY_CIPHER_3DES:
+              log_error (_("WARNING: cipher algorithm '%s' is insecure and not"
+                           " suitable for encryption!\n"),
+                         openpgp_cipher_algo_name (efx->cfx.dek->algo));
+              break;
+          };
+
           make_session_key ( efx->cfx.dek );
           if (DBG_CRYPTO)
             log_printhex (efx->cfx.dek->key, efx->cfx.dek->keylen, "DEK is: ");
